### PR TITLE
File/Module Headers

### PR DESCRIPTION
**Pull request by mariari**

### CODE REVIEW

File: src/Juvix/Frontend.hs [hunk 1]
```diff
@@ -10,22 +9,26 @@ import Juvix.Library
...
-        |> fmap
-          ( Arrow.first
-              (NameSymbol.fromSymbol . intern . toUpper . FilePath.takeBaseName)
-          )
-        |> Right
-        |> pure
+    Right (Types.Header name xs) ->
+      pure (Right (name, xs))
+    Right (Types.NoHeader xs) ->
+      let toName =
```
Comment (cwgoes) [inline src/Juvix/Frontend.hs:29]: This should be top-level?

**Review by cwgoes — CHANGES_REQUESTED** [review]
One nit, and can you explain the intent here? Shouldn't every file have a module name?

**Comment by mariari** [comment]
> One nit, and can you explain the intent here? Shouldn't every file have a module name?

Sure, it basically works like haskell does.

One can give a name for the module at the start or omit it.

File: src/Juvix/Frontend.hs (hunk 1 above)
Comment (cwgoes) [reply src/Juvix/Frontend.hs:29]: Hmm OK I think it's a bit better form but nbd

**Comment by mariari** [comment]
> > > One nit, and can you explain the intent here? Shouldn't every file have a module name?
> > 
> > 
> > Sure, it basically works like haskell does.
> > One can give a name for the module at the start or omit it.
> 
> In which case it will default to the file name, or something like that?

exactly, it works like how it did before we made the change

**Review by cwgoes — APPROVED** [review]
ACK

**Pull request merged**